helper/framing/conn_test.go

```diff
@@ -0,0 +1,292 @@
+package framing_test
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/lomik/go-carbon/helper/framing"
+)
+
+func Test1BigEndian(t *testing.T) {
+	run(t, 1, binary.BigEndian)
+}
+
+func Test2BigEndian(t *testing.T) {
+	run(t, 2, binary.BigEndian)
+}
+
+func Test4BigEndian(t *testing.T) {
+	run(t, 4, binary.BigEndian)
+}
+
+func Test1LittleEndian(t *testing.T) {
+	run(t, 1, binary.LittleEndian)
+}
+
+func Test2LittleEndian(t *testing.T) {
+	run(t, 2, binary.LittleEndian)
+}
+
+func Test4LittleEndian(t *testing.T) {
+	run(t, 4, binary.LittleEndian)
+}
+
+func run(t *testing.T, size byte, endianess binary.ByteOrder) {
+	message := "13 bytes long"
+
+	l, err := net.Listen("tcp", ":0")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer l.Close()
+
+	port := l.Addr().(*net.TCPAddr).Port
+
+	go func() {
+		conn, err := net.Dial("tcp", ":"+strconv.Itoa(port))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		framed, err := framing.NewConn(conn, size, endianess)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer framed.Close()
+
+		for i := 1; i <= 2; i++ {
+			if _, err := fmt.Fprintf(framed, message); err != nil {
+				t.Fatal(err)
+			}
+		}
+	}()
+
+	conn, err := l.Accept()
+	if err != nil {
+		t.Fatal(err)
+	}
+	framed, err := framing.NewConn(conn, size, endianess)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer framed.Close()
+
+	buf, err := framed.ReadFrame()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if msg := string(buf[:]); msg != message {
+		t.Fatalf("Unexpected message:\nGot:\t\t%s\nExpected:\t%s\n", msg, message)
+	}
+
+	fixed := make([]byte, 20) // More than 13
+	n, err := framed.Read(fixed)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if n != 13 {
+		t.Fatal("Frame is not of correct size")
+	}
+	if msg := string(fixed[:13]); msg != message {
+		t.Fatalf("Unexpected message:\nGot:\t\t%s\nExpected:\t%s\n", msg, message)
+	}
+}
+
+func TestInvalidFrameSiz(t *testing.T) {
+	_, err := framing.NewConn(nil, 3, nil)
+	if err != framing.ErrPrefixLength {
+		t.Fail()
+	}
+}
+
+func TestPacketTooLarge(t *testing.T) {
+	max := uint(16)
+	size := byte(4)
+	conn, err := framing.NewConn(nil, size, nil)
+	conn.MaxFrameSize = max
+	if err != nil {
+		t.Fatal(err)
+	}
+	b := make([]byte, max+1)
+	_, err = conn.Write(b)
+	if err != framing.ErrFrameTooLarge {
+		t.Fail()
+	}
+}
+
+func TestMessageSmallerThanBuffer(t *testing.T) {
+	conn := &fake{ReadWriter: bytes.NewBuffer([]byte{4, 0, 1, 2, 3})}
+	framed, _ := framing.NewConn(&fake{ReadWriter: conn}, 1, nil)
+
+	first_b := make([]byte, 6)
```

Comment:
"first_b", "first_n", "first_err", "third_n", "third_err", "framed_conn" and many other variable names are bad. https://golang.org/doc/effective_go.html#mixed-caps